[solarman] Enable write registers

san4esmc opened the pull request:
# [solarman] Add ability to write to Solarman registers

# Description

Current binding implementation doesn't allow to change data in invertor - only reads data from it. At the same time there are sets of registers that can safely be manipulated to automate eletricity usage, etc.
By default all registers are treated as read-only unless other is directly stated in channel configuration, which should avoid any accidental writes.
To make channel writable you should add readOnly=false to channel configuration. I have also added one new invertor definition `deye_sg01hp3` for Deye 30KWh High Volatage inverter, where this was tested. New channels introduced there are connected to Time Of Use functionality, including time periods, battery SOC, charge enabled.

# Testing

Existing unit tests were adapted to refactored code.
To test updated binding you can download built jar here: https://mega.nz/file/jjAlAT4D#mFLoSH1T8NP36iZzJKroslgVaMFkOTtW1ve7hfJ7KnY

stefan-hoehn commented:
I highly appreciate this PR.
Please don't forget to update the Readme.md (including describing new channels).

san4esmc commented:
> I highly appreciate this PR. Please don't forget to update the Readme.md (including describing new channels).

I have updated readme.md now. 
Also added ignoring refresh command (as we use polling to get data), so we don't spam logs with warnings about writable channel not found

stefan-hoehn commented on bundles/org.openhab.binding.solarman/README.md:
```diff
@@ -143,6 +144,18 @@ This is the list you get for the `deye_sg04lp3` inverter type:
 
+## Writable Channels
+
+If you want to write into registers of invertor you need to directly define your channels as writable as by default channels are counted as read only. For example, `deye_sg01hp3` has full set of writable channels connected to Time Of Use functionality, but if you want to add custom writable channels you need to define it like this:
```
of the inverter
read only -> read-only
has full set -> has a full set
`Time Of Use`

san4esmc replied:
Fixed

stefan-hoehn reviewed (changes requested):
I don't see any channel changes in the readme? So no new channels have been added?

Btw, I have SUN-8K-SG04LP3-EU which works well with the binding.

san4esmc commented:
> I don't see any channel changes in the readme? So no new channels have been added?
> 
> Btw, I have SUN-8K-SG04LP3-EU which works well with the binding.

I have added new invertor type where this was tested - `deye_sg01hp3` and in readme.md we have registers example for `deye_sg04lp3`. I can add new registers to that one also, but we'll need someone to test that everything is working there also

stefan-hoehn commented:
I was referring to what you wrote in the description:

>"New channels introduced there are connected to Time Of Use functionality, including time periods, battery SOC, charge enabled."

I might have overlooked them in the Readme? 🤔

lsiepel reviewed (approved):
Thansk for providing this PR. 
LGTM

@stefan-hoehn do you need to finsih some tests?
